etk-asm/src/asm.rs

```diff
@@ -221,21 +211,20 @@ pub struct Assembler {
 
     /// Labels, in `pending`, that have been referred to (ex. with push) but
     /// have not been declared with an `AbstractOp::Label`.
-    undeclared_labels: HashSet<String>,
+    undefined_labels: Vec<PendingLabel>,
```

Comment:
It's been a while, so bear with me, but how do we want to use "undeclared" vs "undefined"?

You're probably more familiar with the code than I am at this point, but IIRC, undefined labels were labels that had a `some_label:` in the source code, but didn't have a fixed position yet because of a `%push` macro. Undeclared labels were forward references to a `some_label:` on a later line.

Reply:
In this case, it's only necessary to keep track of the labels that were pushed before they were declared. I agree with you that the use of the word _undefined_ is not quite correct, so I made the modification so that the field is called _undeclared_labels_. I also corrected the documentation to better understand the meaning of each struct field.